pkg/certs/cert-inspection/certgraphanalysis/metadata_options.go

```diff
@@ -43,4 +52,41 @@ var (
 			}
 		},
 	}
+	RewriteNodeIPs = &metadataOptions{
+		rewriteCertKeyPair: func(certKeyPair *certgraphapi.CertKeyPair, o rewriteParams) {
```

Comment:
I think this construct is pointing to a need to make 

```go
type metadataOptions interface{
  rewriteCABundle(caBundle *certgraphapi.CertificateAuthorityBundle)
  rewriteCertKeyPair(certKeyPair *certgraphapi.CertKeyPair)
}
```

So that you can have a caller do

```go
certgraphanalysis.GatherCertsFromPlatformNamespaces(ctx, kubeClient, 
  certgraphanalysis.SkipRevisioned, 
  certgraphanalysis.ElideProxyCADetails,
  certgraphanalysis.RewriteNodeIPs(nodes),
)
```

by having

```go
func RewriteNodeIPs(nodes []*corev1.Nodes) metadataOptions{
  return &nodeIPRewriter{nodes: nodes}
}
```

Reply:
writing the interface as above will allow us to separate decisions about "what sorts of things should we filter on" from "and now every option needs to change".  And rather than having missing information that is lightly enforced, we can use construction of various options to ensure data is correct and only those callers that want particular filters need to gather the data.

Choosing the nodes instead of a client makes unit testing easier and we have little performance benefit or temporal locality to gain by using a client.